Fixes #9025 - DataTable: RowHover attribute doesn't work

jasonex7 opened the pull request:
Fixes #9025 

It seems this problem was introduced while fixing #8620 specifically in commit https://github.com/primefaces/primefaces/commit/dbf8912705925d13e281bd6ff0c40cd2ead56dc4. I'm creating a PR reverting partially that commit. Maybe @ihsanyapicioglu could review it.

jasonex7 committed:
Fixes #9025 - DataTable: RowHover attribute doesn't work

Revert partially https://github.com/primefaces/primefaces/commit/dbf8912705925d13e281bd6ff0c40cd2ead56dc4

melloware commented:
I think you reverted too much.  When I worked with @ihsanyapicioglu on issue #8620 i thought it should have been this one line fix.

basically its a 1 line fix in datatable.js change...
```js
 var checkboxes = this.tbody.find('> tr.ui-datatable-selectable > td.ui-selection-column > div.ui-chkbox > div.ui-chkbox-box:visible'),
```
to
```js
var checkboxes = this.jq.find('tr.ui-datatable-selectable > td.ui-selection-column > div.ui-chkbox > div.ui-chkbox-box:visible'),
```

it has to find the checkboxes in both the frozen table and regular table and it was only searching through TBODY of the regular table

melloware commented:
What was happening with frozen you can use this reproducer:
[pf-8260.zip](https://github.com/primefaces/primefaces/files/9173646/pf-8260.zip)

When you clicked hte SelectAll checkbox it was not selecting them all

 
![image](https://user-images.githubusercontent.com/4399574/180603327-1bc65017-aa6d-4f6d-89c0-15e977a10f2e.png)

melloware commented on primefaces/src/main/resources/META-INF/resources/primefaces/datatable/datatable.js:
```diff
@@ -2842,7 +2842,7 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         var shouldCheckAll = true;
         if(this.cfg.nativeElements) {
-            var checkboxes = this.jq.find('tr.ui-datatable-selectable > td.ui-selection-column > :checkbox:visible'),
+            var checkboxes = this.tbody.find('tr.ui-datatable-selectable > td.ui-selection-column > :checkbox:visible'),
```
I think you have to leave this line

melloware commented on primefaces/src/main/resources/META-INF/resources/primefaces/datatable/datatable.js:
```diff
@@ -2861,7 +2861,7 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         }
         else {
-            var checkboxes = this.jq.find('tr.ui-datatable-selectable > td.ui-selection-column > div.ui-chkbox > div.ui-chkbox-box:visible'),
+            var checkboxes = this.tbody.find('tr.ui-datatable-selectable > td.ui-selection-column > div.ui-chkbox > div.ui-chkbox-box:visible'),
```
I think you have to leave this line

jasonex7 committed:
Fixes #9025 - DataTable: RowHover attribute doesn't work

Revert partially https://github.com/primefaces/primefaces/commit/dbf8912705925d13e281bd6ff0c40cd2ead56dc4

jasonex7 committed:
Fixes #9025 - DataTable: RowHover attribute doesn't work

Revert partially https://github.com/primefaces/primefaces/commit/dbf8912705925d13e281bd6ff0c40cd2ead56dc4
